cclunit-framework-source/doc/CCLUTMOCKING.md

```diff
@@ -0,0 +1,284 @@
+  
+Supported escape values  
+\\| = | (to represent a pipe in character fields)  
+\\\\ = \ (to represent a backslash in character fields)  
+\null = null (no value will be inserted into the column)  
+  
+@param tableName  
+&nbsp;&nbsp;&nbsp;&nbsp;The name of the source table for the mock table to which the data will be added.  
+@param rowData  
+&nbsp;&nbsp;&nbsp;&nbsp;A pipe-delimited string of data to be inserted into the mock table.  
+  
+Example:  
+call cclutDefineMockTable("person", "person_id|name_last|name_first|birth_dt_tm", "f8|vc|vc|dq8")  
+call cclutCreateMockTable("person")  
+call cclutAddMockData("person", "1.0|Washington|George|01-JAN-1970 00:00") ;Will add George Washington  
+call cclutAddMockData("person", "2.0|A\\|d\\\\ams|John|02-FEB-1971 11:11") ;Will add John A|d\ams  
+call cclutAddMockData("person", "3.0|Jefferson|\null|03-MAR-1972 22:22") ;Will add Jefferson (no first name)  
+call cclutAddMockData("person", "4.0|Madison||04-APR-1973 10:33") ;Will add Madison (empty string for first name)
+
+**cclutClearMockData(tableName = vc)**
+
+Clears all data from the mock table.  This is functionally similar to a truncate.  tableName is required.  The table  
+must have been created through cclutCreateMockTable() or else an error will be thrown.  
+  
+@param tableName  
+&nbsp;&nbsp;&nbsp;&nbsp;The name of the source table for the mock table to be cleared.  
+  
+Example:  
+call cclutClearMockData("person")
+
+**cclutAddMockImplementation(originalName = vc, replaceName = vc)**
+
+Adds a mock implementation to be utilized by cclutExecuteProgramWithMocks.  This can include record structures, subroutines, or other programs.  originalName and replaceName are required.  If originalName is already being mocked, the mock will be overwritten with the new replaceName.  
+  
+@param originalName  
+&nbsp;&nbsp;&nbsp;&nbsp;The object to be mocked.  
+@param replaceName  
+&nbsp;&nbsp;&nbsp;&nbsp;The mocked object.  
+  
+Example:  
+call cclutAddMockImplementation("uar_get_code_by", "mock_uar_get_code_by")
+
+**cclutRemoveMockImplementation(originalName = vc)**
+
+Removes a mock implementation.  
+  
+@param originalName  
+&nbsp;&nbsp;&nbsp;&nbsp;The object that is mocked.  
+  
+Example:  
+call cclutRemoveMockImplementation("uar_get_code_by")
+
+**cclutRemoveAllMockImplementations(null)**
+
+Removes all mock implementations.  
+  
+Example:  
+call cclutRemoveAllMockImplementations(null)
+
+**cclutRemoveAllMocks**
+
+Removes all mock implementations and mock tables that have been added through the cclutAddMockImplementation() and cclutCreateMockTable() APIs.  This should be called at the completion of a test suite to clean up all mocks.  
```

Comment:
It should be "test case" rather than "test suite".

It seems like the warning statement ("cclutRemoveAllMocks should be called at the completion....") is lost in the middle of this file. Perhaps it should be made at the beginning.  ???

One option is to place the cclutRemoveAllMocks documentation right after the documentation for cclutExecuteProgramWithMocks and simply say "through the mocking APIs" or "through the mocking APIs described below"  rather than citing specific functions.

Reply:
Will do.